orchestration/src/main/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatModel.java

```diff
@@ -0,0 +1,71 @@
+package com.sap.ai.sdk.orchestration.spring;
+
+import com.sap.ai.sdk.orchestration.AssistantMessage;
+import com.sap.ai.sdk.orchestration.OrchestrationClient;
+import com.sap.ai.sdk.orchestration.OrchestrationPrompt;
+import com.sap.ai.sdk.orchestration.SystemMessage;
+import com.sap.ai.sdk.orchestration.UserMessage;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.ai.chat.messages.Message;
+import org.springframework.ai.chat.model.ChatModel;
+import org.springframework.ai.chat.model.ChatResponse;
+import org.springframework.ai.chat.prompt.Prompt;
+
+/** Spring AI integration for the orchestration service. */
+@Slf4j
+@RequiredArgsConstructor
+public class OrchestrationChatModel implements ChatModel {
+  @Nonnull private OrchestrationClient client;
+
+  /** Default constructor. */
+  public OrchestrationChatModel() {
+    this.client = new OrchestrationClient();
+  }
+
+  @Nonnull
+  @Override
+  public ChatResponse call(@Nonnull final Prompt prompt) {
+
+    if (prompt.getOptions() != null
+        && prompt.getOptions() instanceof OrchestrationChatOptions options) {
```

Comment:
(Minor)

`instansteof` is already only true for non-null values.

```suggestion
    if (prompt.getOptions() instanceof OrchestrationChatOptions options) {
```